v2/svid/jwtsvid/svid.go

```diff
@@ -57,6 +57,11 @@ func ParseAndValidate(token string, bundles jwtbundle.Source, audience []string)
 			return nil, jwtsvidErr.New("token header missing key id")
 		}
 
+		// forbid tokens which have the `typ` header, which is not either "JOSE" or "JWT"
```

Comment:
nit: we should extract this check into its own function for re-use

Reply:
would you like that done as part of this PR?

Reply:
That would actually be great. We should just move this check the parse function which is called both by ParseAndValidate and ParseInsecure. It can probably be done right after the token is parsed using jwt.ParseSigned.